sharding/utils/marshal.go

```diff
@@ -0,0 +1,185 @@
+}
+
+// serializeBlob parses the blob and serializes it appropriately.
+func serializeBlob(cb interface{}) ([]byte, error) {
+
+	interfaceblob, err := ConvertInterface(cb, reflect.Slice)
+	if err != nil {
+		return nil, fmt.Errorf("Error: %v", err)
+	}
+	blob := interfacetoByte(interfaceblob)
+	length := int64(len(blob))
+	terminalLength := length % chunkDataSize
+	chunksNumber := length / chunkDataSize
+	indicatorByte := make([]byte, 1)
+	indicatorByte[0] = 0
+	tempbody := []byte{}
+
+	// if blob is less than 31 bytes, it adds the indicator chunk and pads the remaining empty bytes to the right
+
+	if chunksNumber == 0 {
+		paddedbytes := make([]byte, (chunkDataSize - length))
+		indicatorByte[0] = byte(terminalLength)
+		tempbody = append(indicatorByte, append(blob, paddedbytes...)...)
+		return tempbody, nil
+	}
+
+	//if there is no need to pad empty bytes, then the indicator byte is added as 00011111
+	// Then this chunk is returned to the main Serialize function
+
+	if terminalLength == 0 {
+
+		for i := int64(1); i < chunksNumber; i++ {
+			// This loop loops through all non-terminal chunks and add a indicator byte of 00000000, each chunk
+			// is created by appending the indcator byte to the data chunks. The data chunks are separated into sets of
+			// 31
+			tempbody = append(tempbody,
+				append(indicatorByte,
+					blob[(i-1)*chunkDataSize:i*chunkDataSize]...)...)
+
+		}
+		indicatorByte[0] = byte(chunkDataSize)
+
+		// Terminal chunk has its indicator byte added, chunkDataSize*chunksNumber refers to the total size of the blob
+		tempbody = append(tempbody,
+			append(indicatorByte,
+				blob[(chunksNumber-1)*chunkDataSize:chunkDataSize*chunksNumber]...)...)
+
+		return tempbody, nil
+
+	}
+
+	// This loop loops through all non-terminal chunks and add a indicator byte of 00000000, each chunk
+	// is created by appending the indcator byte to the data chunks. The data chunks are separated into sets of
+	// 31
+
+	for i := int64(1); i <= chunksNumber; i++ {
+
+		tempbody = append(tempbody,
+			append(indicatorByte,
+				blob[(i-1)*chunkDataSize:i*chunkDataSize]...)...)
+
+	}
+	// Appends indicator bytes to terminal-chunks , and if the index of the chunk delimiter is non-zero adds it to the chunk.
+	// Also pads empty bytes to the terminal chunk.chunkDataSize*chunksNumber refers to the total size of the blob.
+	// finalchunkIndex refers to the index of the last data byte
+	indicatorByte[0] = byte(terminalLength)
+	tempbody = append(tempbody,
+		append(indicatorByte,
+			blob[chunkDataSize*chunksNumber:length]...)...)
+
+	emptyBytes := make([]byte, (chunkDataSize - terminalLength))
+	tempbody = append(tempbody, emptyBytes...)
+
+	return tempbody, nil
+
+}
+
+// Serialize takes a set of blobs and converts them to a single byte array.
+func Serialize(rawtx []interface{}) ([]byte, error) {
+	length := int64(len(rawtx))
+
+	if length == 0 {
+		return nil, fmt.Errorf("Validation failed: Collation Body has to be a non-zero value")
```

Comment:
No validation in this library! If the user wants to serialize an empty set of data then you return an empty set of data. Is it possible that the object that uses this library (a collation) could have an empty body and that is valid.

Reply:
Alright will remove this